D/D.sublime-syntax

```diff
@@ -8,339 +8,2032 @@ file_extensions:
+        2: keyword.operator.lambda.d
+      set: value
+    - match: '{{identifier_ref}}(?=\s*\()'
+      scope: variable.function.d
+      set: value-after
+    - match: '{{identifier_ref}}'
+      scope: variable.other.d
+      set: value-after
+    - include: not-whitespace-illegal
+  cast-after:
+    - match: '\)'
+      scope: punctuation.section.parens.end.d
+      set: value
+    - include: not-whitespace-illegal
+  new-after-parens:
+    - match: '\)'
+      scope: punctuation.section.parens.end.d
+      set: new-after
+    - include: not-whitespace-illegal
+  new-after:
+    - match: '\bclass\b'
+      scope: storage.type.class.d
+      set:
+        - match: '\('
+          scope: punctuation.section.parens.begin.d
+          set: [new-anonymous-after-args, value-list]
+        - match: '(?={)'
+          set: optional-template-body-block
+        - match: '(?=\S)'
+          set: [optional-template-body-block, base-class-list]
+    - match: (?=\S)
+      set: [new-after-named, type]
+    - include: not-whitespace-illegal
+  new-after-named:
+    - match: '\('
+      scope: punctuation.section.parens.begin.d
+      set: [value-parens-after, value-list]
+    - match: (?=\S)
+      set: value-after
+  new-anonymous-after-args:
+    - match: '\)'
+      scope: punctuation.section.parens.end.d
+      set:
+        - match: '(?={)'
+          set: optional-template-body-block
+        - match: '(?=\S)'
+          set: [optional-template-body-block, base-class-list]
+    - include: not-whitespace-illegal
+  is-after-type:
+    - match: '\b({{name}})\b'
+      scope: variable.parameter.d
+    - match: '\)'
+      scope: punctuation.section.parens.end.d
+      pop: true
+    - match: '\=\=|:'
+      scope: keyword.operator.logical.d
+      set: [is-after, type-specialization]
+  is-after:
+    - match: ','
+      scope: punctuation.separator.sequence.d
+      set: function-argument-definition-list-content
+    - match: '\)'
+      scope: punctuation.section.parens.end.d
+      pop: true
+    - include: not-whitespace-illegal
+  type-specialization:
+    - match: '\b(?:({{type_specialization_storage}})|({{type_specialization_keyword}}))\b'
+      captures:
+        1: storage.type.d
+        2: keyword.d
+      set:
+        - match: '(?=\))'
+          pop: true
+        - include: type-specialization
+    - match: '(?=\S)'
+      set: type
+    - include: not-whitespace-illegal-pop
+  value-group-after:
+    - match: '\b(function|delegate)\b'
+      scope: keyword.other.d
+      set: [value-group-after, basic-type2, function-type]
+    - match: '\)'
+      scope: punctuation.section.group.end.d
+      set: value-group-after-parens
+    - match: '\b({{name}})\b'
+      scope: variable.parameter.d
+      set: value-group-after
+    - match: ','
+      scope: punctuation.separator.sequence.d
+      set: [value-group-after-parens, function-argument, function-argument-or-type]
+    - match: '\.\.\.'
+      scope: keyword.operator.d
```

Comment:
Some syntaxes call this `keyword.operator.spread` (JS) others `keyword.operator.variadic` (C, C++, Go, Java). It probably makes sense to adopt the second.